src/Orleans/Configuration/ConfigUtilities.cs

```diff
@@ -525,48 +525,36 @@ internal static string IStatisticsConfigurationToString(IStatisticsConfiguration
         /// <returns>The string representation of the DataConnectionString with account credential info redacted.</returns>
-        public static string RedactConnectionStringInfo(string dataConnectionString)
+        public static string RedactConnectionStringInfo(string connectionString)
         {
-            return PrintSqlConnectionString(
-                PrintDataConnectionInfo(dataConnectionString));
-        }
+            string[] secretKeys =
+            {
+                "AccountKey=",                              // Azure Storage
+                "SharedAccessSignature=",                   // Many Azure services
+                "SharedAccessKey=", "SharedSecretValue=",   // ServiceBus
+                "Password=",                                // SQL
+                "SecretKey=", "SessionToken=",              // DynamoDb
+            };
 
-        /// <summary>Removes private credential information about an azure connection string by truncating the account key from it.</summary>
-        /// <param name="azureConnectionString">The original connection string</param>
-        public static string PrintDataConnectionInfo(string azureConnectionString)
-        {
-            if (String.IsNullOrEmpty(azureConnectionString)) return "null";
+            if (String.IsNullOrEmpty(connectionString)) return "null";
 
-            string azureConnectionInfo = azureConnectionString;
-            // Remove any Azure account keys from connection string info written to log files
-            int accountKeyPos = azureConnectionInfo.LastIndexOf("AccountKey=", StringComparison.Ordinal);
-            if (accountKeyPos > 0)
-            {
-                azureConnectionInfo = azureConnectionInfo.Remove(accountKeyPos) + "AccountKey=<--SNIP-->";
-            }
-            return azureConnectionInfo;
-        }
+            string connectionInfo = connectionString;
 
-        /// <summary>Removes private credential information about a SQL connection string by truncating the account password from it.</summary>
-        /// <param name="sqlConnectionString">The original connection string</param>
-        public static string PrintSqlConnectionString(string sqlConnectionString)
-        {
-            if (String.IsNullOrEmpty(sqlConnectionString))
-            {
-                return "null";
-            }
-            var sqlConnectionInfo = sqlConnectionString;
-            // Remove any Azure account keys from connection string info written to log files
-            int keyPos = sqlConnectionInfo.LastIndexOf("Password=", StringComparison.OrdinalIgnoreCase);
-            if (keyPos > 0)
+            // Remove any secret keys from connection string info written to log files
+            foreach (var secretKey in secretKeys)
             {
-                sqlConnectionInfo = sqlConnectionInfo.Remove(keyPos) + "Password=<--SNIP-->";
+                int keyPos = connectionInfo.IndexOf(secretKey, StringComparison.OrdinalIgnoreCase);
```

Comment:
because I'm using the index value later on to truncate the string. BTW, I didn't change the logic compared to the previous approach, other than I also validate if the index is `>= 0` (it was just `> 0` before, so a secret could escape if it was the first thing in the connection string)